modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerIndexRebuildTest.java

```diff
@@ -0,0 +1,369 @@
+
+        startupTestCluster();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTestsStopped() throws Exception {
+        stopAllGrids();
+
+        cleanPersistenceDir();
+
+        super.afterTestsStopped();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+
+        G.allGrids().forEach(ignite -> assertFalse(((IgniteEx)ignite).context().maintenanceRegistry().isMaintenanceMode()));
+    }
+
+    /** */
+    private void startupTestCluster() throws Exception {
+        for (int i = 0; i < GRIDS_NUM; i++ )
+            startGrid(i);
+
+        IgniteEx ignite = grid(0);
+
+        ignite.cluster().state(ClusterState.ACTIVE);
+
+        createAndFillCache(ignite, CACHE_NAME_1_1, GRP_NAME_1);
+        createAndFillCache(ignite, CACHE_NAME_1_2, GRP_NAME_1);
+        createAndFillCache(ignite, CACHE_NAME_2_1, GRP_NAME_2);
+
+        createAndFillThreeFieldsEntryCache(ignite, CACHE_NAME_NO_GRP, null, Collections.singletonList(complexIndexEntity()));
+
+        assertTrue(grid(LAST_NODE_NUM).context().config().getBuildIndexThreadPoolSize() > 1);
+    }
+
+    /**
+     * Checks error messages when trying to rebuild indexes for non-existent cache or when trying
+     * to rebuild non-existent indexes.
+     */
+    @Test
+    public void testErrors() {
+        injectTestSystemOut();
+
+        IgniteEx lastNode = grid(LAST_NODE_NUM);
+
+        assertEquals(EXIT_CODE_OK, execute("--cache", "indexes_rebuild",
+            "--node-id", lastNode.localNode().id().toString(),
+            "--target", CACHE_NAME_NON_EXISTING));
+
+        String notExistingCacheOutputStr = testOut.toString();
+
+        assertTrue(notExistingCacheOutputStr.contains("WARNING: Indexes rebuild was not scheduled for any cache. Check command input."));
+
+        testOut.reset();
+
+        assertEquals(EXIT_CODE_OK, execute("--cache", "indexes_rebuild",
+            "--node-id", grid(LAST_NODE_NUM).localNode().id().toString(),
+            "--target", CACHE_NAME_1_1 + "=non-existing-index"));
+
+        String notExistingIndexOutputStr = testOut.toString();
+
+        assertTrue(notExistingIndexOutputStr.contains("WARNING: Indexes rebuild was not scheduled for any cache. Check command input."));
+
+        assertTrue(notExistingIndexOutputStr.contains(
+            "WARNING: These indexes were not found:" + System.lineSeparator()
+            + INDENT + CACHE_NAME_1_1 + ":" + System.lineSeparator()
+            + INDENT + INDENT + "non-existing-index")
+        );
+    }
+
+    /**
+     * Checks that index is rebuilt correctly.
+     */
+    @Test
+    public void testRebuild() throws Exception {
+        injectTestSystemOut();
+
+        IgniteEx node = grid(LAST_NODE_NUM);
+
+        assertEquals(EXIT_CODE_OK, execute("--cache", "indexes_rebuild",
+            "--node-id", node.localNode().id().toString(),
+            "--target", CACHE_NAME_NO_GRP));
+
+        node.close();
```

Comment:
Please add a check that an index rebuild is scheduled for the given cache and index.